### PR TITLE
added tests  8a & 8b to do 1D/2D mesh writing and reading

### PR DESCRIPTION
**Pull request by guj**

### CODE REVIEW

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+ * @param numSeg        num of subdivition for the elements
+ *                      note that with h5collect mode, numSeg must be 1
+ * @param numSteps      num of iterations
+ */
+class TestInput
+{
+public:
+  TestInput() =  default;
+
+  unsigned int GetSeg() const
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]: Please add a doxygen string to this function.

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+
+/**     Class TestInput
+ *
+ *
+ * @param mpi_size      MPI size
+ * @param mpi_rank      MPI rank
+ * @param bulk          num of elements
+ * @param numSeg        num of subdivition for the elements
+ *                      note that with h5collect mode, numSeg must be 1
+ * @param numSteps      num of iterations
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]: I might overlook this, but do these parameters belong here? `@param` is usually a function parameter. Let me document below how to describe member variables.

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 3]
```diff
@@ -0,0 +1,651 @@
...
+  unsigned int m_Seg = 1;
+  int m_Steps = 1;
+  int m_TestNum = 0;
+  std::string m_Backend = ".bp";
+  bool m_Unbalance = false;
+
+  int m_Ratio = 1;
+
+  Extent m_GlobalMesh;
+  std::vector<std::pair<unsigned long, unsigned long>> m_InRankDistribution;
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]:
```suggestion
  int m_MPISize = 1; //!< MPI size
  int m_MPIRank = 0; //!< MPI rank
  unsigned long m_Bulk = 1000ul; //!< num of elements
  /** number of subdivisions for the elements
   *
   * note that with h5collect mode, m_Seg must be 1
   */
  unsigned int m_Seg = 1;
  int m_Steps = 1;   //!< num of iterations
  int m_TestNum = 0; //!< ... description here ...
  std::string m_Backend = ".bp"; //!< I/O backend by file ending
  bool m_Unbalance = false;      //!< ... description here ...

  int m_Ratio = 1; //!< ... description here ...

  Extent m_GlobalMesh; //!< ... description here ...
  /** ... description here ...
   */
  std::vector<std::pair<unsigned long, unsigned long>> m_InRankDistribution;
```

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+        input.m_Steps = atoi( argv[4] );
+
+
+    auto backends = getBackends();
+    for (auto which: backends)
+      {
+    input.m_Backend = which;
+    input.run(1);
+    input.run(2);
+      }
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]:
```suggestion
    auto const backends = getBackends();
    for( auto const & which: backends )
    {
        input.m_Backend = which;
        input.run(1);
        input.run(2);
    }
```

File: examples/8b_benchmark_read1D2D_parallel.cpp [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+class MemoryProfiler
+{
+    /**
+     *
+     * Simple Memory profiler for linux
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+public:
```
Comment (ax3l) [inline examples/8b_benchmark_read1D2D_parallel.cpp]:
```suggestion
public:
    /** Simple Memory profiler for linux
     *
     * @param[in] rank     MPI rank
     * @param[in] tag      item name to measure
     */
```

File: examples/8b_benchmark_read1D2D_parallel.cpp [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+  }
+
+    /**
+     *
+     * Read from /proc/self/status and display the Virtual Memory info at rank 0 on console
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+
```
Comment (ax3l) [inline examples/8b_benchmark_read1D2D_parallel.cpp]:
```suggestion
    /** Read from /proc/self/status and display the Virtual Memory info at rank 0 on console
     *
     * @param tag      item name to measure
     */
```

File: CMakeLists.txt [hunk 7]
```diff
@@ -627,6 +627,9 @@ set(openPMD_EXAMPLE_NAMES
     6_dump_filebased_series
     7_extended_write_serial
     8_benchmark_parallel
+    8a_benchmark_write1D2D_parallel
+    8b_benchmark_read1D2D_parallel
+
```
Comment (ax3l) [inline CMakeLists.txt]:
```suggestion
```

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 8]
```diff
@@ -0,0 +1,651 @@
...
+class MemoryProfiler
+{
+    /**
+     *
+     * Simple Memory profiler for linux
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+public:
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]:
```suggestion
public:
    /** Simple Memory profiler for linux
     *
     * @param[in] rank     MPI rank
     * @param[in] tag      item name to measure
     */
```

Comment (ax3l) [reply examples/8a_benchmark_write1D2D_parallel.cpp]: Maybe a `std::string` for the tag can be more C++-ish?

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 9]
```diff
@@ -0,0 +1,651 @@
...
+  }
+
+    /**
+     *
+     * Read from /proc/self/status and display the Virtual Memory info at rank 0 on console
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]:
```suggestion
    /** Read from /proc/self/status and display the Virtual Memory info at rank 0 on console
     *
     * @param[in] tag      item name to measure
     */
```

File: examples/8a_benchmark_write1D2D_parallel.cpp [hunk 10]
```diff
@@ -0,0 +1,651 @@
...
+
+/** The Memory profiler class for profiling purpose
+ *
+ *  Simple Memory usage report that works on linux system
+ */
+
+static std::chrono::time_point<std::chrono::system_clock>  m_ProgStart = std::chrono::system_clock::now();
+
+class MemoryProfiler
```
Comment (ax3l) [inline examples/8a_benchmark_write1D2D_parallel.cpp]: Should  this class go into `include/openPMD/benchmark/`? That would reduce some code duplication in both examples.

File: examples/8b_benchmark_read1D2D_parallel.cpp [hunk 11]
```diff
@@ -0,0 +1,519 @@
...
+    if (argc >= 3)
+      input.m_Pattern = atoi(argv[2]);
+
+    auto backends = getBackends();
+    for (auto which: backends)
+      {
+    input.m_Backend = which;
+    input.run(prefix);
+      }
```
Comment (ax3l) [inline examples/8b_benchmark_read1D2D_parallel.cpp]:
```suggestion
    auto const backends = getBackends();
    for( auto const & which: backends )
    {
        input.m_Backend = which;
        input.run(prefix);
    }
```

File: CMakeLists.txt (hunk 7 above)
Comment (guj) [reply CMakeLists.txt]: ok

**Pull request merged**